walkthroughs/1C-managing-and-protecting-using-api-mgmt/walkthrough.adoc

```diff
@@ -0,0 +1,381 @@
+* link:https://www.3scale.net[3scale Website, window="_blank"]
+****
+
+
+:sectnums:
+
+[time=5]
+== Configure OpenID Connect in Red Hat Single Sign-On
+
+{3scale-ProductName} supports several authentication methods, including a single API key, a pair of Application ID and Application key, and OpenID Connect. The latter is the most secure way to protect APIs.
+We need to configure {sso-ProductName} to allow OpenID Connect authentication on the API gateway.
+
+=== Configure a client for Zync
+
+3scale synchronizes client credentials between 3scale and {sso-ProductName} using a component called Zync. Whenever a new application is created in 3scale, a client is created in the configured {sso-ProductName} realm. A client with special permissions needs to be configured to do this.
+
+. Log in to the link:{user-sso-url}/auth/admin/{user-sanitized-username}/console[{sso-ProductName} administration console, window="_blank"] using `{user-username}` as username and `password` as password.
+
+. Create a new client:
+.. Click *Clients* from the side navigation.
+.. Click *Create* button.
+.. In the *Client ID* field, enter `{zync-client-id}`.
+.. Confirm that `openid-connect` is selected in the *Client Protocol* field.
+.. Click *Save*. You will be redirected to the client configuration screen.
+. Configure client basic settings:
+.. On the *Settings* tab, set the following values for the fields:
+... Set _Access Type_ to `confidential`.
+... Set _Standard Flow Enabled_ to `OFF`.
+... Set _Direct Access Grants Enabled_ to `OFF`.
+... Set _Service Accounts Enabled_ to `ON`.
+... Click *Save* to save the changes.
+. Configure client permissions:
+.. Navigate to *Service Account Roles* tab of the client configuration page.
+.. In the _Client Roles_ drop-down list, select `realm-management`.
+.. In the _Available Roles_ pane, select the `manage-clients` list item and assign the role by clicking *Add selected* button.
+. Get the client secret:
+.. Navigate to the *Credentials* tab of the client configuration page.
+.. Note the value in the *Secret* field (it will be referred to as `ZYNC_SECRET`), it will be used in later steps.
+
+[type=verification]
+Have you configured the `{zync-client-id}` client and noted its Client Secret?
+
+[type=verificationFail]
+Verify that you followed each step in the procedure above. If you are still having issues, contact your administrator.
+
+:sectnums!:
+
+// Task resources go here
+[type=taskResource]
+.Task Resources
+****
+* link:https://access.redhat.com/documentation/en-us/red_hat_3scale_api_management/2.6/html/using_the_developer_portal/openid-connect[OpenID Connect Integration, window="_blank"]
+****
+
+
+:sectnums:
+
+[time=5]
+== Configure 3scale API Management
+
+=== Log in to 3scale API Management
+
+. Open the link:https://{user-sanitized-username}-admin.{openshift-app-host}[{3scale-ProductName} Login screen, window="_blank"].
```

Comment:
Should we add a brief note explaining "this is your personal tenant and is isolated from other users" or do you think the URL makes that pretty clear?

Reply:
Yep, would be good to mention it. Will add.